benchmark_runner/common/remote_ssh/remote_ssh.py

```diff
@@ -0,0 +1,195 @@
+                self.get_remote_file(remote, local, file_name)
+
+    def put_remote_file(self, local, remote, file_name):
+        """
+        This method upload file name from local to remote path
+        :param local:
+        :param remote:
+        :param file_name:
+        :return:
+        """
+        self.__p_sftp.put(f'{local}/{file_name}', f'{remote}/{file_name}')
+
+    def get_remote_file(self, remote, local, file_name):
+        """
+        This method download file name from remote to local path
+        :param local:
+        :param remote:
+        :param file_name:
+        :return:
+        """
+        self.__p_sftp.get(f'{remote}/{file_name}', f'{local}/{file_name}')
+
+    def replace_parameter(self, remote_path, file_name, parameter, value, all_line=False):
+        """
+        This method is replace parameter in remote file
+        :param remote_path: path to file
+        :param file_name: file name
+        :param parameter: the parameter
+        :param value: the new value
+        :param all_line: replace all the line that contains the parameter with value
+        :return:
+        """
+        if all_line:
+            res = False
+            for c in ['/', '@', '#', '$', '%', '^', '&', '*', '-', '<', '>', '!', '~', '+', '-']:
+                if c not in parameter:
+                    res = True
+                    self.run_command(f'sed -i \'s{c}{parameter}{c}{value}{c}g\' {remote_path}/{file_name}')
+                    break
+            if not res:
+                raise RunCommandError
+        else:
+            if os.path.splitext(file_name)[1] == '.xml':
+                self.run_command(f'sed -i \'s/{parameter}[^ ]*"/{parameter}{value}/g\' {remote_path}/{file_name}')
+            else:
+                self.run_command(f'sed -i \'s/{parameter}[^ ]*/{parameter}{value}/g\' {remote_path}/{file_name}')
+
+    def exist(self, remote_path):
+        """
+        This method check if file or dir exists on remote
+        :param: remote_path to file or dir:
+        :return: True if exists otherwise False
+        """
+        try:
+            self.__p_sftp.stat(remote_path)
+            return True
+        except IOError:
+            return False
+
+    def rename(self, old_remote_path, new_remote_path):
+        """
+        This method rename remote dir
+        :param old_remote_path:
+        :param new_remote_path:
+        """
+        self.__p_sftp.rename(old_remote_path, new_remote_path)
+
+    def rmdir(self, remote_path):
+        """
+        This method delete entire remote folder recursively
+        """
+        if self.exist(remote_path) and len(remote_path) > 10:
+            self.run_command(f'rm -r {remote_path}')
+
+    def copy(self, remote_source, remote_target):
+        """
+        This method copy from remote source to target folder
+        """
+        if self.exist(remote_source):
+            self.run_command(f'cp - ar - f {remote_source} {remote_target}')
```

Comment:
Not sure exactly what `- ar - f` means.  Should it be

```
cp -arf --
```

again, in case filenames start with `-`.

Reply:
Hi,

I have changed it to 
self.run_command(f"cp - arf -- \"{remote_source}\" \"{remote_target}\"")

Reply:
Got it.
Changed it to:  self.run_command(f"cp -arf -- '{remote_source}' '{remote_target}'")
e.g. cp -arf -- 'aa/aa.txt' 'bb/bb.txt'